trailmet/algorithms/prune/network_slimming_prune.py

```diff
@@ -0,0 +1,232 @@
+import os
+import torch
+import torch.nn as nn
+from torch.autograd import Variable
+from torchvision import datasets,transforms
+import sys
+sys.path.append("../../../../")
+import numpy as np
+from trailmet.trailmet.models import channel_selection
+import trailmet.trailmet.models as models
+#from ...models import models
+""
+class get_params():
+  def __init__(self,dataset,test_bs,percent,model,save,depth,arch):
+    self.dataset=dataset
+    self.test_bs=test_bs
+    self.percent=percent  # percentage of network to be pruned
+    self.model=model      # path where the model to be pruned is saved
+    self.save=save        # path where pruned model will be saved
+    self.depth=depth      # depth of model if arch is resnet
+    self.arch=arch        # vgg-16 and resnet family is supported
+initial=get_params('cifar10',64,0.6,'/content/drive/MyDrive/VGG-16 NS/CIFAR-10/trained/resnet_model_best.pth.tar','/content/drive/MyDrive/VGG-16 NS/CIFAR-10/pruned/resnet_pruned1.pth.tar',164,'resnet')
+""
+"""
+defining model architecture
+"""
+class prune_it:
```

Comment:
this is not really the format. prune_it seems more of a function and should not be a class. Follow the structure in chipnet.py. Your code is the closest to chipnet and it should be easy to organize accordingly